compiler_gym/service/runtime/CreateAndRunCompilerGymServiceImpl.h

```diff
@@ -51,7 +51,7 @@ void setGrpcChannelOptions(grpc::ServerBuilder& builder);
 //       createAndRunCompilerGymServiceImpl(argc, argv, "usage string");
 //     }
 template <typename CompilationSessionType>
-[[noreturn]] void createAndRunCompilerGymServiceImpl(int argc, char** argv, const char* usage) {
+[[nodiscard]] int createAndRunCompilerGymServiceImpl(int argc, char** argv, const char* usage) {
```

Comment:
It triggers a compiler warning if the user doesn't do anything with the return value. With `-Werror`, it prevents the common typo:
 
```c++
int main(int argc, char** argv) {
  createAndRunCompilerGymServiceImpl(&argc, &argv, usage);
}
```
instead of:

```c++
int main(int argc, char** argv) {
  return createAndRunCompilerGymServiceImpl(&argc, &argv, usage);
}
```